applications: nrf5340_audio: Remove dependencies from error_handler

erikrobstad opened the pull request:
The center LED is now controlled
by the GPIO driver directly

OCT-2463

erikrobstad commented:
One question: 
Should we set the LED if the config is release (if CONFIG_DEBUG=n)?
In this case the system will reboot right after

alexsven commented:
> One question: Should we set the LED if the config is release (if CONFIG_DEBUG=n)? In this case the system will reboot right after

Don't think that should be necessary no

alexsven commented on applications/nrf5340_audio/src/utils/error_handler.c:
```diff
@@ -8,48 +8,43 @@
-#endif /* defined(CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP) */
-}
+/* nRF5340 Audio DK center RGB LED */
+static const struct gpio_dt_spec center_led_r = GPIO_DT_SPEC_GET(DT_NODELABEL(rgb1_red), gpios);
+static const struct gpio_dt_spec center_led_g = GPIO_DT_SPEC_GET(DT_NODELABEL(rgb1_green), gpios);
+static const struct gpio_dt_spec center_led_b = GPIO_DT_SPEC_GET(DT_NODELABEL(rgb1_blue), gpios);
```
This is now tied very heavily to our board. If the pin/pins were passed in with an init function this file could reside in NCS instead.

koffes replied:
That is possible, though, the system may fail before we have a chance to init the error handler.

erikrobstad commented:
Changed so that the center LED is only used if the board is Audio DK.
Also changed from gpio_pin_set_dt() to gpio_pin_configure_dt(), so that the error handler should work even if the LEDs are not configured beforehand.
